### PR TITLE
PIM-10909: Refactor command to remove non-existing products and model…

### PR DESCRIPTION
**Pull request by mmetayer**
…s from ES index

**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

**Review by Beninho — APPROVED** [review]

File: src/Akeneo/Pim/Enrichment/Bundle/Storage/ElasticsearchAndSql/ProductAndProductModel/GetDeletedProductDocumentIds.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            }
+
+            yield $resultsPage;
+        } while (count($resultsPage) > 0);
+    }
+
+    /**
+     * @return string[]
+     */
+    private function getNonExistingProductIds(array $productDocumentIds): array
```
[the remark this answers is not in the document]
Comment (antoinearrive) [reply src/Akeneo/Pim/Enrichment/Bundle/Storage/ElasticsearchAndSql/ProductAndProductModel/GetDeletedProductDocumentIds.php:90]: Effectively, I just added the use of SqlGetProductUuids

File: src/Akeneo/Pim/Enrichment/Bundle/Command/CleanRemovedProductsCommand.php [hunk 2]
```diff
@@ -23,170 +19,60 @@
...
 
         return Command::SUCCESS;
     }
 
-    private function removeDocumentFromIndex(OutputInterface $output, int $batchSize): int
+    private function deleteDocumentsFromIndex(iterable $getDeletedDocumentIds, OutputInterface $output): int
     {
-        $indexedProductCount = 0;
+        $numberOfDeleteDocuments = 0;
+        $documentsIdsToRemove = [];
```
Comment (pierallard) [inline src/Akeneo/Pim/Enrichment/Bundle/Command/CleanRemovedProductsCommand.php:53]: $doumentIdsToRemove

File: tests/back/Pim/Enrichment/Integration/Storage/ElasticsearchAndSql/ProductAndProductModel/GetDeleteProductModelDocumentIdsIntegration.php [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+        $this->get('akeneo_integration_tests.fixture.loader.product_and_product_model_with_removed_attribute')->load();
+
+        for ($i = 0; $i < 30; $i++) {
+            $this->createProductModel(sprintf('product_model_identifier_%s', $i), 'a_family_variant', []);
+        }
+        $this->get('akeneo_elasticsearch.client.product_and_product_model')->refreshIndex();
+    }
+
+    protected function createProductModel(string $code, string $familyVariantCode, array $data): ProductModelInterface
```
Comment (pierallard) [inline tests/back/Pim/Enrichment/Integration/Storage/ElasticsearchAndSql/ProductAndProductModel/GetDeleteProductModelDocumentIdsIntegration.php:65]: Why protected ?

**Review by pierallard — APPROVED** [review]

File: tests/back/Pim/Enrichment/Integration/Storage/ElasticsearchAndSql/ProductAndProductModel/GetDeleteProductModelDocumentIdsIntegration.php (hunk 3 above)
Comment (antoinearrive) [reply tests/back/Pim/Enrichment/Integration/Storage/ElasticsearchAndSql/ProductAndProductModel/GetDeleteProductModelDocumentIdsIntegration.php:65]: it's a mistake

**Pull request merged**